util/src/main/scala/org/broadinstitute/dsde/workbench/util/package.scala

```
@@ -32,4 +40,17 @@ package object util {
 
   def readFile[F[_]](path: String)(implicit sf: Sync[F]): Resource[F, FileInputStream] =
     Resource.make(sf.delay(new FileInputStream(path)))(f => sf.delay(f.close()))
+
+  /*
+   * Example:
+   * scala> org.broadinstitute.dsde.workbench.util.readJsonFileToA[IO, List[String]](java.nio.file.Paths.get("/tmp/list"), None).compile.lastOrError.unsafeRunSync
+   * res0: List[String] = List(this is great)
+   *
+   */
+  def readJsonFileToA[F[_]: Sync: ContextShift: Concurrent, A: Decoder](path: Path, blockingExecutionContext: Option[ExecutionContext] = None): Stream[F, A] = {
```

what is the behavior if the path passed isn't a valid path? Will the stream be 'empty' and it fail silently, or will it return some sort of throwable wrapped in the stream?

---

the `F` will fail with appropriate error like
```
scala> org.broadinstitute.dsde.workbench.util.readJsonFileToA[IO, List[String]](java.nio.file.Paths.get("/tmp/list"), None).compile.lastOrError.unsafeRunSync
java.nio.file.NoSuchFileException: /tmp/list
  at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
  at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
  at sun.nio.fs.UnixException.rethrowAsIOException(
```